Missing fields in ElasticIndex on a clean setup

When I start GN5, connecting it to an empty GN4 (then Elasticsearch initialized but before loading any metadata record) I get this exception:

```
Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'openApiConfiguration': Unsatisfied dependency expressed through field 'dynamicPropertiesFacade': Error creating bean with name 'dynamicPropertiesFacade': Unsatisfied dependency expressed through field 'elasticTypingSystem': Error creating bean with name 'elasticTypingSystem' defined in URL [jar:file:/home/antos/.m2/repository/org/geonetwork/ogcapi-records-shared/5.0.0-SNAPSHOT/ogcapi-records-shared-5.0.0-SNAPSHOT.jar!/org/geonetwork/ogcapi/service/indexConvert/dynamic/ElasticTypingSystem.class]: Failed to instantiate [org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem]: Constructor threw exception
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:787) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:767) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:145) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:508) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1421) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:599) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:962) ~[spring-context-6.1.10.jar:6.1.10]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:624) ~[spring-context-6.1.10.jar:6.1.10]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.1.jar:3.3.1]
        at org.geonetwork.GeonetworkApplication.main(GeonetworkApplication.java:24) ~[classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dynamicPropertiesFacade': Unsatisfied dependency expressed through field 'elasticTypingSystem': Error creating bean with name 'elasticTypingSystem' defined in URL [jar:file:/home/antos/.m2/repository/org/geonetwork/ogcapi-records-shared/5.0.0-SNAPSHOT/ogcapi-records-shared-5.0.0-SNAPSHOT.jar!/org/geonetwork/ogcapi/service/indexConvert/dynamic/ElasticTypingSystem.class]: Failed to instantiate [org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem]: Constructor threw exception
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:787) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:767) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:145) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:508) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1421) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:599) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:784) ~[spring-beans-6.1.10.jar:6.1.10]
        ... 20 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'elasticTypingSystem' defined in URL [jar:file:/home/antos/.m2/repository/org/geonetwork/ogcapi-records-shared/5.0.0-SNAPSHOT/ogcapi-records-shared-5.0.0-SNAPSHOT.jar!/org/geonetwork/ogcapi/service/indexConvert/dynamic/ElasticTypingSystem.class]: Failed to instantiate [org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem]: Constructor threw exception
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:306) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1357) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1194) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:784) ~[spring-beans-6.1.10.jar:6.1.10]
        ... 34 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem]: Constructor threw exception
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:221) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:111) ~[spring-beans-6.1.10.jar:6.1.10]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:315) ~[spring-beans-6.1.10.jar:6.1.10]
        ... 47 common frames omitted
Caused by: java.lang.IllegalStateException: Mapping for top-level field 'creationYearForResource' not found in index 'gn-records'. Available top-level fields: [serviceType, isAboveThreshold, rating, geom, uuid, resourceLanguage, isPublishedToAll, draft, useLimitation, changeDate, isValidInspire, linkUrl, resolutionScaleDenominator, resourceTemporalExtentDetails, id, tag, indexingError, recordGroup, metadataIdentifier, recordLink, shape, hasInspireTheme, format, hasxlinks, otherConstraints, bbox_xmax, documentStandard, measure, dateStamp, harvesterId, isTemplate, spatialRepresentationType, isHarvested, xlink, schema, bbox_xmin, recordOwner, isOpenData, pointOfTruthURL, document, link, validReport, anytext, mainLanguage, recordOperateOn, harvesterUuid, groupPublishedId, recordJoin, bbox_ymax, resourceDate, sourceCatalogue, groupPublished, specificationConformance, popularity, root, scope, featureTypes, coordinateSystem, accessConstraints, email, hasOverview, otherLanguage, processSteps, allKeywords, owner, indexingErrorMsg, isSchemaValid, statusWorkflow, groupOwner, docType, isValid, resourceTypeNameObject, mdStatus, overviewUrl, hasBoundingPolygon, any, parentUuid, resolutionDistance, organisationName, resourceIdentifier, location, bbox_ymin, userSavedCount, linkProtocol, resourceType]
        at org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem.computeElasticType(ElasticTypingSystem.java:155) ~[ogcapi-records-shared-5.0.0-SNAPSHOT.jar:5.0.0-SNAPSHOT]
        at org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem.computeRawElasticTypes(ElasticTypingSystem.java:136) ~[ogcapi-records-shared-5.0.0-SNAPSHOT.jar:5.0.0-SNAPSHOT]
        at org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem.getIndexInfo(ElasticTypingSystem.java:123) ~[ogcapi-records-shared-5.0.0-SNAPSHOT.jar:5.0.0-SNAPSHOT]
        at org.geonetwork.ogcapi.service.indexConvert.dynamic.ElasticTypingSystem.<init>(ElasticTypingSystem.java:111) ~[ogcapi-records-shared-5.0.0-SNAPSHOT.jar:5.0.0-SNAPSHOT]
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[na:na]
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:208) ~[spring-beans-6.1.10.jar:6.1.10]
        ... 49 common frames omitted
```

--- a/config/application-ogcapi-records.yml
+++ b/config/application-ogcapi-records.yml
@@ -5,13 +5,6 @@ geonetwork:
       includeElasticJson: false
     ogcapi-property-mapping:
       fields:
-        - ogcProperty: spatialRepresentationType
-          elasticProperty: cl_spatialRepresentationType.key
-          indexRecordProperty: codelists.["cl_spatialRepresentationType"].[*].properties.["key"]
-          typeOverride: LIST_STRING
-          facetsConfig:
-            - facetName: spatialRepresentationType
-              facetType: TERM
         - ogcProperty: resourceType
           elasticProperty: resourceType
           indexRecordProperty: resourceType
@@ -36,30 +29,6 @@ geonetwork:
             - facetName: keywords
               facetType: TERM
               bucketCount: 50
-        - ogcProperty: updateFrequency
-          elasticProperty: cl_maintenanceAndUpdateFrequency.key
-          indexRecordProperty: codelists.["cl_maintenanceAndUpdateFrequency"].[*].properties.["key"]
-          typeOverride: SINGLE_STRING
-          isSortable: true
-          isQueryable: true
-          facetsConfig:
-            - facetName: updateFrequency
-              facetType: TERM
-              bucketCount: 50
-        - ogcProperty: creationYearForResource
-          elasticProperty: creationYearForResource
-          indexRecordProperty: resourceDateDetails.["creationYearForResource"].[0]
-          isSortable: true
-          facetsConfig:
-            - facetName: creationYearForResource
-              facetType: HISTOGRAM_FIXED_INTERVAL
-              numberBucketInterval: 5
-              bucketCount: 25
-              bucketSorting: VALUE
-            - facetName: creationYearForResource2
-              facetType: HISTOGRAM_FIXED_BUCKET_COUNT
-              bucketCount: 5
-              bucketSorting: VALUE
         - ogcProperty: linkProtocol
           elasticProperty: linkProtocol
           indexRecordProperty: linkProtocols
@@ -71,26 +40,6 @@ geonetwork:
                   filterEquationCql: linkProtocol = 'OGC:WFS'
                 - filterName: availableInViewService
                   filterEquationCql: linkProtocol = 'OGC:WMS'
-        - ogcProperty: createDate
-          elasticProperty: createDate
-          indexRecordProperty: createDate
-          isSortable: true
-          isQueryable: true
-          facetsConfig:
-            - facetName: createDate
-              facetType: HISTOGRAM_FIXED_INTERVAL
-              calendarIntervalUnit: MONTH
-              bucketCount: 25
-              bucketSorting: VALUE
-            - facetName: createDate2
-              facetType: HISTOGRAM_FIXED_BUCKET_COUNT
-              bucketCount: 5
-              bucketSorting: VALUE
-        - ogcProperty: resourceTitleObject
-          elasticProperty: resourceTitleObject.default
-          indexRecordProperty: resourceTitle.["default"]
-          isSortable: true
-          isQueryable: true
         - ogcProperty: resolutionScaleDenominator
           elasticProperty: resolutionScaleDenominator
           indexRecordProperty: resolutionScaleDenominator
@@ -106,22 +55,73 @@ geonetwork:
               facetType: HISTOGRAM_FIXED_BUCKET_COUNT
               bucketCount: 5
               bucketSorting: VALUE
-        - ogcProperty: orgForResource
-          elasticProperty: OrgForResourceObject.default
-          indexRecordProperty: orgForResourceByRole.["OrgForResourceObject"].[*].["default"]
-          facetsConfig:
-            - facetName: orgForResource
-              facetType: TERM
-              bucketCount: 50
-
-
+        ## Commented to avoid issue #116
+        ## You can uncomment following lines after metadata schema initialization and some example record creation
+        # - ogcProperty: spatialRepresentationType
+        #   elasticProperty: cl_spatialRepresentationType.key
+        #   indexRecordProperty: codelists.["cl_spatialRepresentationType"].[*].properties.["key"]
+        #   typeOverride: LIST_STRING
+        #   facetsConfig:
+        #     - facetName: spatialRepresentationType
+        #       facetType: TERM
+        # - ogcProperty: updateFrequency
+        #   elasticProperty: cl_maintenanceAndUpdateFrequency.key
+        #   indexRecordProperty: codelists.["cl_maintenanceAndUpdateFrequency"].[*].properties.["key"]
+        #   typeOverride: SINGLE_STRING
+        #   isSortable: true
+        #   isQueryable: true
+        #   facetsConfig:
+        #     - facetName: updateFrequency
+        #       facetType: TERM
+        #       bucketCount: 50
+        # - ogcProperty: creationYearForResource
+        #   elasticProperty: creationYearForResource
+        #   indexRecordProperty: resourceDateDetails.["creationYearForResource"].[0]
+        #   isSortable: true
+        #   facetsConfig:
+        #     - facetName: creationYearForResource
+        #       facetType: HISTOGRAM_FIXED_INTERVAL
+        #       numberBucketInterval: 5
+        #       bucketCount: 25
+        #       bucketSorting: VALUE
+        #     - facetName: creationYearForResource2
+        #       facetType: HISTOGRAM_FIXED_BUCKET_COUNT
+        #       bucketCount: 5
+        #       bucketSorting: VALUE
+        # - ogcProperty: createDate
+        #   elasticProperty: createDate
+        #   indexRecordProperty: createDate
+        #   isSortable: true
+        #   isQueryable: true
+        #   facetsConfig:
+        #     - facetName: createDate
+        #       facetType: HISTOGRAM_FIXED_INTERVAL
+        #       calendarIntervalUnit: MONTH
+        #       bucketCount: 25
+        #       bucketSorting: VALUE
+        #     - facetName: createDate2
+        #       facetType: HISTOGRAM_FIXED_BUCKET_COUNT
+        #       bucketCount: 5
+        #       bucketSorting: VALUE
+        # - ogcProperty: resourceTitleObject
+        #   elasticProperty: resourceTitleObject.default
+        #   indexRecordProperty: resourceTitle.["default"]
+        #   isSortable: true
+        #   isQueryable: true
+        # - ogcProperty: orgForResource
+        #   elasticProperty: OrgForResourceObject.default
+        #   indexRecordProperty: orgForResourceByRole.["OrgForResourceObject"].[*].["default"]
+        #   facetsConfig:
+        #     - facetName: orgForResource
+        #       facetType: TERM
+        #       bucketCount: 50
 
 
     links:
       base-path: /ogcapi-records
       # Where is GN5's ogcapi-records located?
       # should end in "/" and shouldn't have "//" in the path
-      ogcApiRecordsBaseUrl: ${GN5_BASE_URL:http:////localhost:7979/geonetwork}${geonetwork.openapi-records.links.base-path}/
+      ogcApiRecordsBaseUrl: ${GN5_BASE_URL:http:////localhost:7979/geonetwork}/${geonetwork.openapi-records.links.base-path}/
       # How to link to GN (can be either GN4 or GN5's GN4 proxy)
       # should end in "/" and shouldn't have "//" in the path
       gnBaseUrl: ${GN4_PROXY_BASE_URL:http:////localhost:7979/geonetwork}/

--- a/docker/docker-compose-base.yml
+++ b/docker/docker-compose-base.yml
@@ -33,6 +33,15 @@ x-geonetwork-environment:
     -Des.password=
     -Dgeonetwork.ESFeaturesProxy.targetUri=http://elasticsearch:9200/gn-features/{_}
 
+
+  GEONETWORK_DB_TYPE: postgres
+  GEONETWORK_DB_HOST: database
+  GEONETWORK_DB_PORT: ${GEONETWORK_DB_PORT}
+  GEONETWORK_DB_NAME: ${GEONETWORK_DB_NAME}
+  GEONETWORK_DB_USERNAME: ${GEONETWORK_DB_USERNAME}
+  GEONETWORK_DB_PASSWORD: ${GEONETWORK_DB_PASSWORD}
+  VIRTUAL_HOST: localhost
+
 x-service-geonetwork:
   &default-service-geonetwork
   image: geonetwork:4.4.7

--- a/src/modules/ogcapi-records/ogcapi-records-integration-tests/src/test/java/org/geonetwork/infrastructure/ElasticPgMvcBaseTest.java
+++ b/src/modules/ogcapi-records/ogcapi-records-integration-tests/src/test/java/org/geonetwork/infrastructure/ElasticPgMvcBaseTest.java
@@ -39,7 +39,7 @@ import org.testcontainers.utility.MountableFile;
 
 @SpringBootTest(classes = GeonetworkApplication.class)
 @AutoConfigureMockMvc
-@ActiveProfiles(value = {"test"})
+@ActiveProfiles(value = {"test", "integration-test"})
 @ContextConfiguration(initializers = ElasticPgMvcBaseTest.class)
 public class ElasticPgMvcBaseTest implements ApplicationContextInitializer<ConfigurableApplicationContext> {
 

--- a/src/modules/ogcapi-records/ogcapi-records-integration-tests/src/test/resources/application-integration-test.yml
+++ b/src/modules/ogcapi-records/ogcapi-records-integration-tests/src/test/resources/application-integration-test.yml
@@ -1,0 +1,115 @@
+geonetwork:
+  openapi-records:
+    ogcapi-records-json:
+      includeRawXML: false
+      includeElasticJson: false
+    ogcapi-property-mapping:
+      fields:
+        - ogcProperty: resourceType
+          elasticProperty: resourceType
+          indexRecordProperty: resourceType
+          isSortable: true
+          isQueryable: true
+          facetsConfig:
+            - facetName: resourceType
+              facetType: TERM
+        - ogcProperty: format
+          elasticProperty: format
+          indexRecordProperty: formats
+          addPropertyToOutput: false
+          facetsConfig:
+            - facetName: format
+              facetType: TERM
+              bucketCount: 50
+        - ogcProperty: keywords
+          elasticProperty: tag.langeng
+          indexRecordProperty: tag.[*].["default"]
+          addPropertyToOutput: false
+          facetsConfig:
+            - facetName: keywords
+              facetType: TERM
+              bucketCount: 50
+        - ogcProperty: linkProtocol
+          elasticProperty: linkProtocol
+          indexRecordProperty: linkProtocols
+          facetsConfig:
+            - facetName: availableInServices
+              facetType: FILTER
+              filters:
+                - filterName: availableInDownloadService
+                  filterEquationCql: linkProtocol = 'OGC:WFS'
+                - filterName: availableInViewService
+                  filterEquationCql: linkProtocol = 'OGC:WMS'
+        - ogcProperty: resolutionScaleDenominator
+          elasticProperty: resolutionScaleDenominator
+          indexRecordProperty: resolutionScaleDenominator
+          isSortable: true
+          isQueryable: true
+          facetsConfig:
+            - facetName: resolutionScaleDenominator
+              facetType: HISTOGRAM_FIXED_INTERVAL
+              numberBucketInterval: 100000
+              bucketCount: 25
+              bucketSorting: VALUE
+            - facetName: resolutionScaleDenominator2
+              facetType: HISTOGRAM_FIXED_BUCKET_COUNT
+              bucketCount: 5
+              bucketSorting: VALUE
+        - ogcProperty: spatialRepresentationType
+          elasticProperty: cl_spatialRepresentationType.key
+          indexRecordProperty: codelists.["cl_spatialRepresentationType"].[*].properties.["key"]
+          typeOverride: LIST_STRING
+          facetsConfig:
+            - facetName: spatialRepresentationType
+              facetType: TERM
+        - ogcProperty: updateFrequency
+          elasticProperty: cl_maintenanceAndUpdateFrequency.key
+          indexRecordProperty: codelists.["cl_maintenanceAndUpdateFrequency"].[*].properties.["key"]
+          typeOverride: SINGLE_STRING
+          isSortable: true
+          isQueryable: true
+          facetsConfig:
+            - facetName: updateFrequency
+              facetType: TERM
+              bucketCount: 50
+        - ogcProperty: creationYearForResource
+          elasticProperty: creationYearForResource
+          indexRecordProperty: resourceDateDetails.["creationYearForResource"].[0]
+          isSortable: true
+          facetsConfig:
+            - facetName: creationYearForResource
+              facetType: HISTOGRAM_FIXED_INTERVAL
+              numberBucketInterval: 5
+              bucketCount: 25
+              bucketSorting: VALUE
+            - facetName: creationYearForResource2
+              facetType: HISTOGRAM_FIXED_BUCKET_COUNT
+              bucketCount: 5
+              bucketSorting: VALUE
+        - ogcProperty: createDate
+          elasticProperty: createDate
+          indexRecordProperty: createDate
+          isSortable: true
+          isQueryable: true
+          facetsConfig:
+            - facetName: createDate
+              facetType: HISTOGRAM_FIXED_INTERVAL
+              calendarIntervalUnit: MONTH
+              bucketCount: 25
+              bucketSorting: VALUE
+            - facetName: createDate2
+              facetType: HISTOGRAM_FIXED_BUCKET_COUNT
+              bucketCount: 5
+              bucketSorting: VALUE
+        - ogcProperty: resourceTitleObject
+          elasticProperty: resourceTitleObject.default
+          indexRecordProperty: resourceTitle.["default"]
+          isSortable: true
+          isQueryable: true
+        - ogcProperty: orgForResource
+          elasticProperty: OrgForResourceObject.default
+          indexRecordProperty: orgForResourceByRole.["OrgForResourceObject"].[*].["default"]
+          facetsConfig:
+            - facetName: orgForResource
+              facetType: TERM
+              bucketCount: 50

--- a/src/shared/ogcapi-records/src/main/java/org/geonetwork/ogcapi/service/indexConvert/dynamic/ElasticTypingSystem.java
+++ b/src/shared/ogcapi-records/src/main/java/org/geonetwork/ogcapi/service/indexConvert/dynamic/ElasticTypingSystem.java
@@ -148,8 +148,13 @@ public class ElasticTypingSystem {
 
         // start off getting the first level property
         var pathPart = path.removeFirst();
-        PropertyVariant elasticProperty = (PropertyVariant)
-                this.elasticIndexInfo.mappings().properties().get(pathPart)._get();
+        var first = this.elasticIndexInfo.mappings().properties().get(pathPart);
+        if (first == null) {
+            throw new IllegalStateException("Mapping for top-level field '" + pathPart + "' not found in index '"
+                    + indexRecordName + "'. Available top-level fields: "
+                    + this.elasticIndexInfo.mappings().properties().keySet());
+        }
+        PropertyVariant elasticProperty = (PropertyVariant) first._get();
 
         while (!path.isEmpty()) {
             pathPart = path.removeFirst();

